probe/websocket/ws_test.go

```diff
@@ -50,6 +51,13 @@
 		}
 	}()
 
+	// wait for http server to start
+	conn, err := net.DialTimeout("tcp", "localhost:18188", 30*time.Second)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer conn.Close()
```

Comment:
The `conn` object is not used after being created. If its only purpose is to check if the server is up, consider closing it immediately after checking the error instead of deferring its closure. This would free up resources as soon as they are no longer needed. [medium]